### PR TITLE
Add dossier and subdossier link in task listings

### LINKED ISSUE #3013: Aufgabenauflistungen die Spalte "Dossier" verlinken
Müsste ähnlich wie bei Dokumenten gelöst werden, siehe https://github.com/4teamwork/opengever.core/blob/033e68ae4de64908e89cd7726cf516f26ec5ce8b/opengever/document/browser/redirect_to_parent_dossier.py#L9

- [x] Spalte "Dossier" von Tasklistings verlinken
- [x] Spalte "Subdossier" von Tasklistings verlinken

### PR DESCRIPTION
**Pull request by elioschmutz**
Add links to the containing main dossier and the subdossier of a task in task-listings.

bildschirmfoto 2017-06-22 um 17 09 44

closes #3013

### CODE REVIEW

File: opengever/tabbedview/helper.py [hunk 1]
```diff
@@ -101,7 +102,12 @@ def linked_containing_subdossier(item, value):
     if not subdossier_title:
         return ''
 
-    url = "{}/redirect_to_parent_dossier".format(item.getURL())
+    if ICatalogBrain.providedBy(item):
+        url_method = item.getURL
+    else:
+        url_method = item.absolute_url
```
Comment (phgross) [inline opengever/tabbedview/helper.py]: i would add a comment here that you expecting an SQL Task.

File: opengever/tabbedview/helper.py [hunk 2]
```diff
@@ -102,16 +102,15 @@ def linked_containing_subdossier(item, value):
...
-        url_method = item.absolute_url
-
-    url = "{}/redirect_to_parent_dossier".format(url_method())
     title = escape_html(subdossier_title)
 
+    url = get_url(item)
+    if not url:
+        return title
+
+    redirect_url = "{}/redirect_to_parent_dossier".format(get_url(item))
```
Comment (phgross) [inline opengever/tabbedview/helper.py]: Thats a potential XSS security vulnerability, the task title is a user input, please doublecheck.

File: opengever/task/browser/configure.zcml [hunk 3]
```diff
@@ -34,4 +34,10 @@
       permission="zope2.View"
       />
 
+  <browser:page
+      name="redirect_to_parent_dossier"
```
Comment (phgross) [inline opengever/task/browser/configure.zcml]: redirect_to_parent_dossier -> redirect_to_containing_dossier

File: opengever/tabbedview/helper.py [hunk 4]
```diff
@@ -114,6 +114,19 @@ def linked_containing_subdossier(item, value):
...
 
+def linked_containing_maindossier(item, value):
+    title = escape_html(item.containing_dossier)
+
+    url = get_url(item)
+    if not url:
+        return title
+
+    redirect_url = "{}/redirect_to_main_dossier".format(get_url(item))
+    link = '<a href="{}" title="{}" class="maindossierLink">{}</a>'.format(
```
Comment (phgross) [inline opengever/tabbedview/helper.py]: Same here XSS potential ...

File: opengever/tabbedview/helper.py (hunk 2 above)
Comment (elioschmutz) [inline opengever/tabbedview/helper.py]: see line 105: title = escape_html(subdossier_title)

that should prevent for XSS-attacks

File: opengever/tabbedview/helper.py (hunk 4 above)
Comment (elioschmutz) [inline opengever/tabbedview/helper.py]: see line 118: title = escape_html(subdossier_title)

**Comment by elioschmutz** [comment]
@phgross FYI

**Comment by phgross** [comment]
@elioschmutz tests are failing ...

**Comment by elioschmutz** [comment]
@phgross Please take another look!

**Pull request merged**